app/assets/javascripts/discourse/components/search-advanced-options.js.es6

```diff
@@ -74,107 +86,204 @@ export default Em.Component.extend({
   },
 
   _update() {
-    let searchTerm = this.get('searchTerm');
-
-    if (!searchTerm) {
+    if (!this.get('searchTerm')) {
       this._init();
       return;
     }
 
-    this.findUsername(searchTerm);
-    this.findCategory(searchTerm);
-    this.findGroup(searchTerm);
-    this.findBadge(searchTerm);
-    this.findTags(searchTerm);
-    this.findIn(searchTerm);
-    this.findStatus(searchTerm);
-    this.findPostsCount(searchTerm);
-    this.findPostTime(searchTerm);
+    this.findSearchTerms();
+    this.setSearchedTermValue('searchedTerms.username', REGEXP_USERNAME_PREFIX);
+    this.setSearchedTermValueForCategory();
+    this.setSearchedTermValueForGroup();
+    this.setSearchedTermValueForBadge();
+    this.setSearchedTermValueForTags();
+    this.setSearchedTermValue('searchedTerms.in', REGEXP_IN_PREFIX, REGEXP_IN_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.likes', REGEXP_SPECIAL_IN_LIKES_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.private', REGEXP_SPECIAL_IN_PRIVATE_MATCH);
+    this.setSearchedTermSpecialInValue('searchedTerms.special.in.wiki', REGEXP_SPECIAL_IN_WIKI_MATCH);
+    this.setSearchedTermValue('searchedTerms.status', REGEXP_STATUS_PREFIX);
+    this.setSearchedTermValueForPostTime();
+    this.setSearchedTermValue('searchedTerms.posts_count', REGEXP_POST_COUNT_PREFIX);
   },
 
-  findSearchTerm(EXPRESSION, searchTerm) {
+  findSearchTerms() {
+    const searchTerm = this.get('searchTerm');
     if (!searchTerm)
-      return "";
+      return [];
 
-    const expressionPosition = searchTerm.search(EXPRESSION);
-    if (expressionPosition === -1)
-      return "";
+    let result = [];
+    const blocks = searchTerm.match(REGEXP_BLOCKS);
+    _.each(blocks, function(block) {
+      if (block.length === 0) return;
 
-    const remainingPhrases = searchTerm.substring(expressionPosition + 2);
-    let nextExpressionPosition = remainingPhrases.search(REGEXP_FILTER_PREFIXES);
-    if (nextExpressionPosition === -1)
-      nextExpressionPosition = remainingPhrases.length;
+      result.push(block);
+    });
 
-    return searchTerm.substring(expressionPosition, nextExpressionPosition + expressionPosition + 2).trim().split(' ')[0];
+    this.set('searchTermBlocks', result);
   },
 
-  findUsername(searchTerm) {
-    const match = this.findSearchTerm(REGEXP_USERNAME_PREFIX, searchTerm);
+  filterBlocks(regexPrefix) {
+    let result = [];
+    _.each(this.get('searchTermBlocks'), function(block) {
+      if (block.search(regexPrefix) !== -1)
+        result.push(block);
+    });
 
-    if (match.length !== 0) {
-      let userInput = match.replace(REGEXP_USERNAME_PREFIX, '');
+    return result;
+  },
+
+  setSearchedTermValue(key, replaceRegEx, matchRegEx = null) {
+    matchRegEx = matchRegEx || replaceRegEx;
```

Comment:
Um... I don't think that would do what it needs to do. As that would only give you one regex to use in that area, and I definitely need two for one of the conditions. Now if you meant that for `setSearchedTermSpecialInValue`, then yea, I see where I could remove that extra parameter, that was a copy/paste blunder.